go/vt/tabletserver/tabletserver.go

```diff
@@ -244,9 +264,13 @@ const (
 )
 
 // SetServingType changes the serving type of the tabletserver. It starts or
-// stops internal services as deemed necessary.
-func (tsv *TabletServer) SetServingType(tabletType topodatapb.TabletType, serving bool) error {
-	action, err := tsv.decideAction(tabletType, serving)
+// stops internal services as deemed necessary. The tabletType determines the
+// primary serving type, while alsoAllow specifies other tablet types that
+// should also be honored for serving.
+func (tsv *TabletServer) SetServingType(tabletType topodatapb.TabletType, serving bool, alsoAllow []topodatapb.TabletType) error {
+	defer tsv.ExitLameduck()
```

Comment:
I am not sure if it is better to exit lameduck automatically here, or if it should always be the caller's responsibility to call exitlameduck.